### DISCUSSION TITLE
Need different validations for Save as draft and Submit button in form.

### DISCUSSION THREAD

**Discussion by amanchanna** [react-hook-form/react-hook-form#2944] [Q&A]
I have a form having Save as draft and Submit button. I am using schema based validations using Yup. The validations for Save as draft and Submit button are different.

 Basically the user can save as draft even if the required fields are not filled but other validations like max length and amount format etc. should be checked while doing Save as draft.
All the validations including the required one should be applied when doing Submit.
For e.g - We have an amount field that is required and should not be of length more than 14. The validations should be as follows:
For Submit:
-------------
amount: yup.string().required().max(14)

For Save as Draft:
-------------------
amount: yup.string().max(14)

Is there a way we can have different schema for different button click or any other better way to handle this condition in react hook form?

**Comment by bluebill1049** [comment]
why not just use `getValues()` to retrieve all the input values.

**Reply by amanchanna** [reply]
Are you suggesting that we remove the validation schema from form and then perform validation on button click? 
The filed validations should occur on 'onChange' event.

**Reply by bluebill1049** [reply]
no i mean, you can have a button just do the save draft


`<button onClick={async () => await fetch('', {data: getValues()}}>Save draft</button>`

**Reply by amanchanna** [reply]
But then the other validations apart from the required field validation won't apply on Save as Draft.

**Reply by bluebill1049** [reply]
so you still want to validate the form, use resolver and keep two validation schema then.

**Reply by amanchanna** [reply]
I could not find anywhere in documentation on how to apply multiple validation schema and how to switch between these schema for Submit and save as draft.
Can you please guide me through?

**Reply by bluebill1049** [reply]
https://react-hook-form.com/api#useForm

- resolver
- context

**Reply by amanchanna** [reply]
Do you have any example where context is used along with schema?

**Reply by bluebill1049** [reply]
i don't, have a play with this CSB tho: https://codesandbox.io/s/react-hook-form-customresoliver-v6-h4ti6?file=/src/index.jsx

**Comment by sunnyfeijen** [comment]
@amanchanna did you ever solve this?

**Reply by albinbrengesjo** [reply]
Did you find another solution for this? @sunnyfeijen

**Reply by sunnyfeijen** [reply]
@albinbrengesjo I did find a working solution, I've written a bit about in [this dev.to post](https://dev.to/feijens/react-hook-form-multiple-yup-validation-schemas-5a22)

I hope this helps you!

**Comment by albinbrengesjo** [comment]
Is this problem still unsolved? Im also facing this particular issue.

**Comment by sega057** [comment]
Hello!
This is how I implemented it with zod validation:

```
export const draftResolver = zodResolver<zod.ZodType<Partial<YourPayloadType>>>(
    zod.object({
        id: zod.string({
            required_error: 'id is required',
        }),
    }),
);

export const defaultResolver = zodResolver<zod.ZodType<YourPayloadType>>(
...

// Inside a component
const isDraft = React.useRef(false);

const methods = useForm({
    resolver: (...args) => {
        if (isDraft.current) {
            return draftResolver(...args);
        }
        return defaultResolver(...args);
    },
    defaultValues,
});

const onSubmit = (draft?: boolean) => {
    isDraft.current = Boolean(draft);

    methods.handleSubmit(async (data) => {
        // any logic you want to submit the form
    })();
};
```
